web_timeline/static/src/js/timeline_controller.js

```diff
@@ -63,17 +63,20 @@ odoo.define('web_timeline.TimelineController', function (require) {
 
             var fields = this.renderer.fieldNames;
             fields = _.uniq(fields.concat(n_group_bys));
-            self._rpc({
-                model: self.model.modelName,
-                method: 'search_read',
-                kwargs: {
-                    fields: fields,
-                    domain: domains,
-                },
-                context: self.getSession().user_context,
-            }).then(function(data) {
-                return self.renderer.on_data_loaded(data, n_group_bys, defaults.adjust_window);
-            });
+            return $.when(
+                res,
+                self._rpc({
```

Comment:
Not blocking: wouldn't make sense to move the RPC call to a separated func in case a custom behavior is needed?
BTW you could also wrap all the calls into `$.when`, like `$.when(res, self._call_rpc, self._update_rendering)` and `_update_rendering` (random names here) would accept both results of previous calls and then call `on_data_loaded`. The whole chain in this case would become modular.

Reply:
Do you mean `$.when(res, this._loadRecords).then(this._updateRenderer)` where _updateRenderer will accept the results of res, and this._loadRecords?

If this the case this it can be done but then I've to refactor many things, because they're dependent on variable n_group_bys, defaults, and domain.
I think it's better to do it in a different PR with all things that need a refactor.

With this PR I just want to fix the issue.